### PR TITLE
adds redux store to navbar

### PR DESCRIPTION
**Pull request by daveism**
Annoyed the navbar state was not retaining. I updated the navbar to use redux. Commit includes a few minor adjustments to the styling of in-between (large and small) screen sizes that were annoying me.

### CODE REVIEW

**Review by seanebum — APPROVED** [review]

File: src/components/NavBar/NavBar.js [hunk 1]
```diff
@@ -46,14 +48,30 @@ const useStyles = makeStyles((theme) => ({
...
-  // current active tag
-  // TODO: needs persist in redux
-  const [currentTab, setCurrentTab] = React.useState('Home');
+  // redux store for the active tab
+  const dispatch = useDispatch();
+  const navBar = useSelector(NavBarSelector);
+
+  // set active tab for page refreshes and copy past of url
+  useEffect(() => {
+    // get patch to check if path is different from state.
```
Comment (jeffbliss) [inline src/components/NavBar/NavBar.js:61]: should patch be path?

File: src/components/NavBar/NavBar.js [hunk 2]
```diff
@@ -46,14 +48,30 @@ const useStyles = makeStyles((theme) => ({
...
+    // this will happen when a user copy and pastes or uses a shareurl
+    // the tool will need to change the the tab accordingly
+    const locationPath = window.location.pathname.replace('/', '');
+
+    if (!navBar || navBar !== locationPath) {
+      dispatch(changeActiveTab(locationPath));
+    }
+  }, [dispatch, navBar]);
```
Comment (jeffbliss) [inline src/components/NavBar/NavBar.js:69]: why is useEffect state dependent on the dispatch hook?

Comment (daveism) [reply src/components/NavBar/NavBar.js:69]: following the lint rule =`React Hook useEffect has a missing dependency: 'dispatch'. Either include it or remove the dependency array `

Comment (daveism) [reply src/components/NavBar/NavBar.js:69]: fixed in last commit

Comment (daveism) [reply src/components/NavBar/NavBar.js:69]: will keep but we can keep this open for discussion

**Pull request merged**